apollo-federation/src/connectors/json_selection/methods/public/and.rs

```diff
@@ -0,0 +1,332 @@
+        assert!(
+            result.1[0]
+                .message()
+                .contains("Method ->and can only be applied to boolean values.")
+        );
+    }
+
+    #[test]
+    fn and_should_return_error_when_no_arguments_provided() {
+        let result = selection!("$.a->and").apply_to(&json!({
+            "a": true,
+        }));
+
+        assert_eq!(result.0, None);
+        assert!(!result.1.is_empty());
+        assert!(
+            result.1[0]
+                .message()
+                .contains("Method ->and requires arguments")
+        );
+    }
+}
+
+#[cfg(test)]
+mod shape_tests {
+    use shape::location::Location;
+
+    use super::*;
+    use crate::connectors::json_selection::lit_expr::LitExpr;
+
+    fn get_location() -> Location {
+        Location {
+            source_id: SourceId::new("test".to_string()),
+            span: 0..7,
+        }
+    }
+
+    fn get_shape(args: Vec<WithRange<LitExpr>>, input: Shape) -> Shape {
+        let location = get_location();
+        and_shape(
+            &WithRange::new("and".to_string(), Some(location.span)),
+            Some(&MethodArgs { args, range: None }),
+            input,
+            Shape::none(),
+            &IndexMap::default(),
+            &location.source_id,
+        )
+    }
+
+    #[test]
+    fn and_shape_should_return_bool_on_valid_booleans() {
+        assert_eq!(
+            get_shape(
+                vec![WithRange::new(LitExpr::Bool(false), None)],
+                Shape::bool([])
+            ),
+            Shape::bool([get_location()])
+        );
+    }
+
+    #[test]
+    fn and_shape_should_error_on_non_boolean_input() {
+        assert_eq!(
+            get_shape(
+                vec![WithRange::new(LitExpr::Bool(true), None)],
+                Shape::string([])
+            ),
+            Shape::error(
+                "Method ->and can only be applied to boolean values.".to_string(),
+                [get_location()]
+            )
+        );
+    }
+
+    #[test]
+    fn and_shape_should_error_on_non_boolean_args() {
+        assert_eq!(
+            get_shape(
+                vec![WithRange::new(LitExpr::String("test".to_string()), None)],
+                Shape::bool([])
+            ),
+            Shape::error(
+                "Method ->and can only accept boolean arguments.".to_string(),
+                [get_location()]
+            )
+        );
+    }
+
+    #[test]
+    fn and_shape_should_error_on_no_args() {
+        assert_eq!(
+            get_shape(vec![], Shape::bool([])),
+            Shape::error(
+                "Method ->and requires at least one argument".to_string(),
+                [get_location()]
+            )
+        );
+    }
+
+    #[test]
+    fn and_shape_should_error_on_none_args() {
+        let location = get_location();
+        assert_eq!(
+            and_shape(
+                &WithRange::new("and".to_string(), Some(location.span)),
+                None,
```

Comment:
What if an argument's shape computes as `None`? Can we write a test for that? I think it will already produce an error, as desired, but I'm not 100% sure.